cryptography/src/transcript.rs

```diff
@@ -0,0 +1,107 @@
+use rand_core::{CryptoRngCore, OsRng};
+
+/// This struct provides a convenient abstraction over hashing data and deriving randomness.
+///
+/// It automatically takes care of details like:
+/// - correctly segmenting packets of data,
+/// - domain separating different uses of tags and randomness,
+/// - making sure that secret state is zeroized as necessary.
+struct Transcript {}
+
+impl Transcript {
```

Comment:
should this be a `trait` with an associated type `Summary`?

Reply:
I'm skeptical of making this a trait.

Transcripts are not expected to interoperate with other libraries; they're more of a tool to make protocols. For example, in the `stream` case, the result of the transcript is ultimately just used to seed the keys you use for encryption. That you derive a particular string of bytes from a particular input doesn't matter.

It's non-trivial to implement this for multiple different backing implementations, and given the discussion above, I don't see a reason _why_ there should be multiple implementations, so I'd say sticking with a concrete impl that's well understood is a better idea.